tests/manage/z_cluster/nodes/test_disk_failures.py

```diff
@@ -169,87 +171,165 @@ def test_recovery_from_volume_deletion(self, nodes, pvc_factory, pod_factory):
+        ocp.OCP().exec_oc_cmd(
+            f"scale --replicas=0 deployment/{osd_deployment_name}"
+        )
+
+        # Force delete OSD pod if necessary
+        osd_pod_name = osd_pod.name
+        logger.info(f"Waiting for OSD pod {osd_pod.name} to get deleted")
+        if osd_pod.ocp.wait_for_resource(
+            condition=constants.STATUS_TERMINATING, resource_name=osd_pod_name
+        ):
+            osd_pod.delete(force=True)
+        osd_pod.ocp.wait_for_delete(resource_name=osd_pod_name)
+
+        # Run ocs-osd-removal job
+        logger.info(f"Executing OSD removal job on OSD-{osd_id}")
+        osd_removal_job_yaml = ocp.OCP().exec_oc_cmd(
+            f"process -n openshift-storage ocs-osd-removal"
+            f" -p FAILED_OSD_ID={osd_id} -o yaml"
+        )
+        osd_removal_job = OCS(**osd_removal_job_yaml)
+        osd_removal_job.create(do_reload=False)
+
+        # Get ocs-osd-removal pod name
+        logger.info("Getting the ocs-osd-removal pod name")
+        osd_removal_pod_name = get_pod_name_by_pattern(
+            f"ocs-osd-removal-{osd_id}"
+        )[0]
+        osd_removal_pod_obj = get_pod_obj(osd_removal_pod_name)
+        osd_removal_pod_obj.ocp.wait_for_resource(
+            condition=constants.STATUS_COMPLETED,
+            resource_name=osd_removal_pod_name
+        )
+
+        # Verify OSD removal from the ocs-osd-removal pod logs
+        logger.info(
+            f"Verifying removal of OSD from {osd_removal_pod_name} pod logs"
+        )
+        logs = get_pod_logs(osd_removal_pod_name)
+        pattern = f"purged osd.{osd_id}"
+        assert re.search(pattern, logs)
 
         # Delete the OSD prepare job
+        logger.info(f"Deleting OSD prepare job {osd_prepare_job_name}")
         osd_prepare_job.delete()
-        osd_prepare_job.ocp.wait_for_delete(resource_name=osd_prepare_job_name, timeout=120)
+        osd_prepare_job.ocp.wait_for_delete(
+            resource_name=osd_prepare_job_name, timeout=120
+        )
 
         # Delete the OSD PVC
         osd_pvc_name = osd_pvc.name
         logger.info(f"Deleting OSD PVC {osd_pvc_name}")
         osd_pvc.delete()
         osd_pvc.ocp.wait_for_delete(resource_name=osd_pvc_name)
 
-        # Recreate a volume from the platform side
-        logger.info("Creating a replacing volume from the platform side")
-        nodes.create_and_attach_volume(osd_node, volume_size)
+        # Delete the OSD deployment
+        osd_deployment_name = osd_deployment.name
+        logger.info(f"Deleting OSD deployment {osd_deployment_name}")
+        osd_deployment.delete()
+        osd_deployment.ocp.wait_for_delete(
+            resource_name=osd_deployment_name, timeout=120
+        )
+
+        # Delete PV
+        logger.info(f"Verifying deletion of PV {osd_pv_name}")
+        if osd_pv.ocp.wait_for_resource(
+            condition="Failed", resource_name=osd_pv_name
+        ):
```

Comment:
This will raise CommandFailed exception if the PV is already deleted.

Reply:
Handled the exception in the latest commit